src/components/loading/LoadingActions.ts

```diff
@@ -25,16 +26,20 @@ export const removeLoading = (id: string): IReduxAction<ILoadingActionPayload> =
   }
 });
 
-export const turnOnLoading = (ids: string[]): IReduxAction<ILoadingActionPayload> => ({
+// the second id is the id from which the action was triggered from
+export const turnOnLoading = (ids: string[], id?: string): IReduxAction<ILoadingActionPayload> => ({
   type: LoadingActions.turnOn,
   payload: {
-    ids
+    ids,
+    id,
   }
 });
 
-export const turnOffLoading = (ids: string[]): IReduxAction<ILoadingActionPayload> => ({
+// the second id is the id from which the action was triggered from
+export const turnOffLoading = (ids: string[], id?: string): IReduxAction<ILoadingActionPayload> => ({
```

Comment:
Maybe you could change the parameters name to be clearer

`export const turnOffLoading = (loadingIDs: string[], triggeredFromId?: string)`

Reply:
thanks I like that